hudi-common/src/main/java/org/apache/hudi/common/model/HoodieRecordPayload.java

```diff
@@ -146,6 +146,24 @@ default Comparable<?> getOrderingValue() {
     return 0;
   }
 
+  /**
+   * This method can be used to enable projection push down for the payload to trim columns.
+   * @return true if projection push down is enabled, false otherwise.
+   */
+  @PublicAPIMethod(maturity = ApiMaturityLevel.EVOLVING)
+  default boolean enableProjectionPushDown() {
+    return false;
+  }
+
```

Comment:
We better not add interfaces on the payload class, they have been deprecated.

Reply:
> We better not add interfaces on the payload class, they have been deprecated.

Ok, payload mode will no longer be supported in 1.x version, I will close this pr~